code/modules/mob/living/basic/pets/dog.dm

```diff
@@ -750,48 +766,58 @@ GLOBAL_LIST_INIT(strippable_corgi_items, create_strippable_list(list(
 	icon_state = "void_puppy"
 	icon_living = "void_puppy"
 	icon_dead = "void_puppy_dead"
-	nofur = TRUE
+	can_be_shaved = FALSE
 	held_state = "void_puppy"
 	unsuitable_atmos_damage = 0
 	minimum_survivable_temperature = TCMB
 	maximum_survivable_temperature = T0C + 40
 
 /mob/living/basic/pet/dog/corgi/puppy/void/Initialize(mapload)
 	. = ..()
-	ADD_TRAIT(src, TRAIT_AI_BAGATTACK, INNATE_TRAIT)
+	//void puppies can spacewalk and can harass people from within a container because they're void puppies
+	add_traits(list(TRAIT_AI_BAGATTACK, TRAIT_SPACEWALK), INNATE_TRAIT)
 
-/mob/living/basic/pet/dog/corgi/puppy/void/Process_Spacemove(movement_dir = 0, continuous_move = FALSE)
-	return 1 //Void puppies can navigate space.
+//Less exciting dog species
```

Comment:
make a new file for pugs, and also one file for corgi. should be like

```txt
code/modules/mob/basic/pets/dog
| _dog.dm (all root dog code)
| corgis.dm (because it's like 500 lines it's okay to snowflake)
| strippable_items.dm (this could be in _dog.dm, unsure)
| dog_commands.dm (this can be in _dog.dm if you want)
| dog_subtypes.dm (pugs and bullterriers)
```

long files always make my head spin and make these PRs with gargantuan diffs just that harder to review so would be greatly appreciated.